src/protocol/mod.rs

```diff
@@ -449,9 +457,14 @@ impl WebSocketContext {
                     } else {
                         // The server MUST close the connection upon receiving a
                         // frame that is not masked. (RFC 6455)
-                        return Err(Error::Protocol(
-                            "Received an unmasked frame from client".into(),
-                        ));
+                        // The only exception here is if the user explicitly accepts given
+                        // stream (by tungstenite::server::accept_with_config or tungstenite::server::accept_hdr_with_config)
+                        // with WebSocketConfig.server_allow_unmasked set to true
```

Comment:
I would not mention the `accept_with_config()` and `accept_hdr_with_config()` explicitly, I think the option and its usage would be self-explanatory at this place (after all it's documented in the config).

Reply:
comments incorporated, including renaming the flag to _accept_unmasked_frames_ as suggested. I think _accept_unmasked_frames_ is more explanatory in terms of action but less explanatory in terms of who is doing the action. The original name (server_allow_unmasked) was the opposite case. Ideal would be _server_accept_unmasked_frames_ but that is already too long.

Reply:
> but less explanatory in terms of who is doing the action

For the client the option "accept unmasked frames" does not quite make sense, because the client always receives unmasked frames.